backend/src/main/java/se/chalmers/dat265/group1/api/order/OrderController.java

```diff
@@ -36,34 +42,43 @@ public Order getOrder(String orderID) {
     }
 
     @Override
-    public Order createNewOrder(Order order) {
+    public Order createNewOrder(Order order) throws InvalidDateFormatException {
+        validateDate(order.getDate());
         int id = orderDataRepository.postObject(new OrderData(order)).getId();
         for (OrderedPart orderedPart : order.getOrderedParts()) {
-            orderedPart.setID(id);
+            orderedPart.setOrderID(id);
             orderedPartRepository.postObject(orderedPart);
         }
         return getOrder(id + "");
     }
 
     @Override
-    public Order updateOrder(String orderID, Order order) {
+    public Order updateOrder(String orderID, Order order) throws InvalidDateFormatException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'se.chalmers.dat265.group1.api.order.InvalidDateFormatException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)